File: warmups/01.two-sum/main.py
from typing import List

class Solution(object):
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        m = {}

        for i, num in enumerate(nums):
            if m.get(num) == None:
                m[target - num] = i
            else:
                return [ i, m[num] ]
        
        # It is gauranteed that the problem would have one solution,
        # so this line will never be reached
        return [ -1, -1 ]

File: warmups/01.two-sum/main.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int, int> m;

        for(int i = 0; i < n; i++) {
            if (m.find(nums[i]) == m.end()) {
                m.insert({target - nums[i], i});
            } else {
                return { m[nums[i]], i };
            }
        }

        // It is gauranteed that the problem would have one solution,
        // so this line will never be reached
        return { -1, -1 };
    }
};
